src/main/generic/utils/crypto/CryptoUtils.js

```diff
@@ -80,17 +83,44 @@ class CryptoUtils {
      * @return {Promise.<Uint8Array>}
      */
     static async encryptOtpKdf(data, key) {
-        if (data.length !== CryptoUtils.ENCRYPTION_INPUT_SIZE) throw new Error('Invalid data size for encryption');
+        if (data.length !== CryptoUtils.ENCRYPTION_INPUT_SIZE && data.length !== CryptoUtils.IMAGEWALLET_INPUT_SIZE) {
+            throw new Error('Invalid data size for encryption');
+        }
 
-        const salt = new Uint8Array(CryptoUtils.ENCRYPTION_SALT_LENGTH);
-        CryptoWorker.lib.getRandomValues(salt);
+        const type = data.length === CryptoUtils.IMAGEWALLET_INPUT_SIZE ? 3 : 2;
+
+        let buf;
+        switch (type) {
+            case 2: { // Argon2 KDF, Hash checksum
+                const salt = new Uint8Array(CryptoUtils.ENCRYPTION_SALT_LENGTH);
+                CryptoWorker.lib.getRandomValues(salt);
+
+                buf = new SerialBuffer(CryptoUtils.ENCRYPTION_SIZE);
+                buf.writeUint8(2); // type
+                buf.writeUint8(Math.log2(CryptoUtils.ENCRYPTION_KDF_ROUNDS));
+                buf.write(await CryptoUtils.otpKdf(data, key, salt, CryptoUtils.ENCRYPTION_KDF_ROUNDS));
+                buf.write(salt);
+                buf.write(Hash.computeBlake2b(data).subarray(0, CryptoUtils.ENCRYPTION_CHECKSUM_LENGTH));
+                break;
+            }
+            case 3: { // 3 = Imagewallet standard
+                const salt = new Uint8Array(CryptoUtils.IMAGEWALLET_SALT_LENGTH);
+                CryptoWorker.lib.getRandomValues(salt);
+
+                const ciphertext = new SerialBuffer(CryptoUtils.IMAGEWALLET_CIPHERTEXT_LENGTH);
+                ciphertext.write(data);
+                ciphertext.write(Hash.computeBlake2b(data).subarray(0, CryptoUtils.IMAGEWALLET_CHECKSUM_LENGTH));
```

Comment:
@styppo Should the checksum be computed as a Blake2b hash in the image wallet standard?
Perhaps a more standard hash algorithm could be better. Otherwise everyone using the image wallet needs Blake2B.